pkg/genericapiserver/config.go

```diff
@@ -71,6 +71,7 @@ type Config struct {
 	// allow downstream consumers to disable the index route
 	EnableIndex           bool
 	EnableProfiling       bool
+	EnableVersion         bool
```

Comment:
Very frustrating since this wrong in every use-case except one.

@nikhiljindal what about making the `Version` route accept a struct and having the server that's actually using it provide the route?  We still wouldn't use it downstream, but at least it would clutter up the genericapiserver.